File: Python/백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys
input = sys.stdin.readline

while True:
    a, b, c = map(int, input().split())
    if a == 0 and b == 0 and c == 0:
        break

    # 정렬: 가장 긴 변을 맨 마지막에
    sides = sorted([a, b, c])
    x, y, z = sides  # z는 가장 긴 변

    if x + y <= z:
        print("Invalid")
    else:
        if x == y == z:
            print("Equilateral")
        elif x == y or y == z or x == z:
            print("Isosceles")
        else:
            print("Scalene")

File: Java/백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String line;
        while ((line = br.readLine()) != null) {
            StringTokenizer st = new StringTokenizer(line);
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int c = Integer.parseInt(st.nextToken());
            if (a == 0 && b == 0 && c == 0) {
                break;
            }

            int[] sides = new int[]{a, b, c};
            Arrays.sort(sides);
            int x = sides[0], y = sides[1], z = sides[2];

            if (x + y <= z) {
                System.out.println("Invalid");
            } else {
                if (x == y && y == z) {
                    System.out.println("Equilateral");
                } else if (x == y || y == z || x == z) {
                    System.out.println("Isosceles");
                } else {
                    System.out.println("Scalene");
                }
            }
        }
        br.close();
    }
}